### PR TITLE
Multipart update

### PR DESCRIPTION
**Pull request by msyyc**
bases on https://github.com/allenjzhang/typespec-e2e-demo/pull/32

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
regen of java
https://github.com/msyyc/typespec-e2e-demo/pull/1

File: todoApp/spec/main.tsp [hunk 1]
```diff
@@ -275,9 +287,18 @@ namespace TodoItems {
...
+
+   @sharedRoute
+    @post
+    op createFileAttachment(
+      @header contentType: "multipart/form-data",
       @path itemId: TodoItem.id,
-      @body contents: TodoAttachment,
-    ): WithStandardErrors<NoContentResponse | NotFoundErrorResponse>;
+      @multipartBody body: FileAttachmentMultipartRequest,
+    ): WithStandardErrors<NoContentResponse | NotFoundResponse>;
```
Comment (MaryGao) [inline todoApp/spec/main.tsp]: Maybe no NotFoundResponse? https://github.com/allenjzhang/typespec-e2e-demo/pull/32#discussion_r1884385919

Comment (msyyc) [reply todoApp/spec/main.tsp]: So we shall still keep `NotFoundErrorResponse`

**Review by MaryGao — APPROVED** [review]
LGTM

**Review by lirenhe — APPROVED** [review]

**Pull request merged**